src/volue/mesh/tests/test_availability.py

```diff
@@ -155,64 +163,1294 @@ def test_add_revision_recurrence_and_get_event(session):
+        targets=[THERMAL_COMPONENT_PATH],
+    )
+
+    # Assert we found at least one restriction
+    assert len(results) > 0
+
+    # Find our specific restriction in results
+    found = False
+    for result in results:
+        if result.event_id == "searchable_restriction_event":
+            found = True
+            assert result.local_id == "searchable_restriction_local_id"
+            assert result.reason == "Searchable restriction reason"
+            assert result.category == "DischargeMin[m3/s]"
+            break
+
+    assert found, "Created restriction not found in search results"
+
+    # Get the restriction directly
+    retrieved_restriction = session.availability.get_availability_event(
+        target=THERMAL_COMPONENT_PATH,
+        event_id="searchable_restriction_event",
+    )
+
+    # Verify it matches what we created
+    assert retrieved_restriction.event_id == "searchable_restriction_event"
+    assert retrieved_restriction.local_id == "searchable_restriction_local_id"
+    assert retrieved_restriction.reason == "Searchable restriction reason"
+    assert retrieved_restriction.category == "DischargeMin[m3/s]"
+
+    # Verify recurrence details
+    assert isinstance(retrieved_restriction.recurrence, RestrictionBasicRecurrence)
+
+
+@pytest.mark.database
+def test_search_instances(session):
+    # Create a revision with a recurrence that repeats daily for 5 days
+    revision = session.availability.create_revision(
+        target=THERMAL_COMPONENT_PATH,
+        event_id="search_instances_event",
+        local_id="search_instances_local_id",
+        reason="Search instances test",
+    )
+
+    # Add a repeating recurrence
+    session.availability.add_revision_recurrence(
+        target=THERMAL_COMPONENT_PATH,
+        event_id=revision.event_id,
+        period_start=datetime(2023, 1, 1, 8, 0, tzinfo=dateutil.tz.UTC),  # 8 AM
+        period_end=datetime(2023, 1, 1, 16, 0, tzinfo=dateutil.tz.UTC),  # 4 PM
+        recurrence=Recurrence(
+            status="Planned",
+            description="Daily Recurrence for Search Test",
+            recurrence_type=RecurrenceType.DAILY,
+            recur_every=1,
+            recur_until=datetime(
+                2023, 1, 5, 23, 59, tzinfo=dateutil.tz.UTC
+            ),  # For 5 days
+        ),
+    )
+
+    # Search for all instances in a 3-day period
+    search_start = datetime(2023, 1, 2, 0, 0, tzinfo=dateutil.tz.UTC)  # Day 2
+    search_end = datetime(2023, 1, 4, 23, 59, tzinfo=dateutil.tz.UTC)  # Day 4
+
+    instances = session.availability.search_instances(
+        target=THERMAL_COMPONENT_PATH,
+        event_id=revision.event_id,
+        period_start=search_start,
+        period_end=search_end,
+    )
+
+    # We should have 3 instances (for days 2, 3, and 4)
+    assert len(instances) == 3
+
+    expected_day = 2
+    # There is a bug in mesh that makes the instances in reverse order
+    # So we need to reverse the list to check the expected order
```

Comment:
Add TODO: so we can find it easier and reduce the likelihood of forgetting to update the test once fixed in Mesh?